### ISSUE TITLE
[NFR]: PHP 8.2 support

### ISSUE THREAD

**Issue by Jeckerson** [phalcon/cphalcon#15973]
~Blocked by https://github.com/zephir-lang/zephir/issues/2370~

**Comment by andypost** [comment]
Used to build 5.0_rc4 for Alpinelinux, just a small patch allowed to build without errors

https://gitlab.alpinelinux.org/alpine/aports/-/merge_requests/37407/diffs#b79d2b5ff56a34ca9289565a23e8b9f62d3db676

**Comment by andypost** [comment]
It still fails to build on PHP 8.2 with a huge log of warnings and notices
```
...
/usr/include/php82/Zend/zend_portability.h:364:52: note: in definition of macro 'UNEXPECTED'
  364 | # define UNEXPECTED(condition) __builtin_expect(!!(condition), 0)
      |                                                    ^~~~~~~~~
/usr/include/php82/Zend/zend_API.h:1944:9: note: in expansion of macro 'Z_PARAM_STR_EX'
 1944 |         Z_PARAM_STR_EX(dest, 0, 0)
      |         ^~~~~~~~~~~~~~
/home/buildozer/aports/testing/php82-pecl-phalcon/src/phalcon-5.0.0/phalcon.zep.c:218942:17: note: in expansion of macro 'Z_PARAM_STR'
218942 |                 Z_PARAM_STR(key)
       |                 ^~~~~~~~~~~
/usr/include/php82/Zend/zend_API.h:2106:76: note: expected 'zend_string **' {aka 'struct _zend_string **'} but argument is of type 'zval *' {aka 'struct _zval_struct *'}
 2106 | static zend_always_inline bool zend_parse_arg_str(zval *arg, zend_string **dest, bool check_null, uint32_t arg_num)
      |                                                              ~~~~~~~~~~~~~~^~~~
/home/buildozer/aports/testing/php82-pecl-phalcon/src/phalcon-5.0.0/phalcon.zep.c: In function 'zim_Phalcon_Annotations_Adapter_Stream_write':
/usr/include/php82/Zend/zend_API.h:1937:58: warning: passing argument 2 of 'zend_parse_arg_str' from incompatible pointer type [-Wincompatible-pointer-types]
 1937 |                 if (UNEXPECTED(!zend_parse_arg_str(_arg, &dest, check_null, _i))) { \
/usr/include/php82/Zend/zend_portability.h:364:52: note: in definition of macro 'UNEXPECTED'
  364 | # define UNEXPECTED(condition) __builtin_expect(!!(condition), 0)
      |                                                    ^~~~~~~~~
/usr/include/php82/Zend/zend_API.h:1944:9: note: in expansion of macro 'Z_PARAM_STR_EX'
 1944 |         Z_PARAM_STR_EX(dest, 0, 0)
      |         ^~~~~~~~~~~~~~
/home/buildozer/aports/testing/php82-pecl-phalcon/src/phalcon-5.0.0/phalcon.zep.c:219024:17: note: in expansion of macro 'Z_PARAM_STR'
219024 |                 Z_PARAM_STR(key)
       |                 ^~~~~~~~~~~
/usr/include/php82/Zend/zend_API.h:2106:76: note: expected 'zend_string **' {aka 'struct _zend_string **'} but argument is of type 'zval *' {aka 'struct _zval_struct *'}
 2106 | static zend_always_inline bool zend_parse_arg_str(zval *arg, zend_string **dest, bool check_null, uint32_t arg_num)
      |                                                              ~~~~~~~~~~~~~~^~~~
/home/buildozer/aports/testing/php82-pecl-phalcon/src/phalcon-5.0.0/phalcon.zep.c: In function 'php_zephir_init_globals':
/home/buildozer/aports/testing/php82-pecl-phalcon/src/phalcon-5.0.0/phalcon.zep.c:220757:57: warning: assignment to 'zend_string *' {aka 'struct _zend_string *'} from incompatible pointer type 'char *' [-Wincompatible-pointer-types]
220757 |         phalcon_globals->orm.resultset_prefetch_records = ZSTR_VAL(zend_string_init(ZEND_STRL("0"), 0));
       |                                                         ^
make: *** [Makefile:210: phalcon.lo] Error 1
```

See https://build.alpinelinux.org/buildlogs/build-edge-x86_64/testing/php82-pecl-phalcon/php82-pecl-phalcon-5.0.0-r0.log

**Comment by Jeckerson** [comment]
@andypost Yes, Zephir changes are required. Use this branch of Zephir to build and compile Phalcon - https://github.com/zephir-lang/zephir/tree/%232370-php8.2

**Comment by v00v** [comment]
@Jeckerson link you provided is broken, it shows error page 404

**Comment by hakimio** [comment]
@v00v I guess he was referring to [this PR](https://github.com/zephir-lang/zephir/pull/2392) which was merged into [zephir development branch](https://github.com/zephir-lang/zephir/tree/development).

**Comment by andypost** [comment]
Yes, and there's new release https://github.com/zephir-lang/zephir/releases/tag/0.17.0

**Comment by andypost** [comment]
Used to build from https://github.com/phalcon/cphalcon/archive/44119cf8b3b286b15ad7d6405ad9e1f443d3bc09.tar.gz on Alpinelinux, it's required `export CFLAGS="$CFLAGS -DALPINE_LINUX=1"` to skip execinfo miss according to https://github.com/zephir-lang/zephir/issues/2397

**Comment by Jeckerson** [comment]
Implemented in #16282
